### PR TITLE
Merge 20.5 code freeze into `trunk`

### PR DESCRIPTION
**Pull request by mokagio**
- [x] Switch to production Pods
- [x] New version header in `RELEASE-NOTE.txt`
- [x] `Localizable.strings` updated
- [x] `release_notes.txt` updated with notes from `RELEASE-NOTE.txt` for current version
- [x] Version update in `.xcconfig`

### CODE REVIEW

File: Podfile.lock
Comment (AliSoftware) [inline Podfile.lock:606]: @mokagio I saw Slack notifications telling that `WordPressAuthenticator 2.1.0` had been published to CocoaPods a couple of minutes after `WordPressKit 4.57.0` has. So I'm wondering if it's expected that you still point to WPAuthenticator 2.0.0 here? (well, more precisely, that this `~> 2.0.0` here was still resolved to exactly `2.0.0` and not `2.1.0` on line 495 of this lockfile)

File: WordPress/Jetpack/Resources/release_notes.txt
Comment (AliSoftware) [inline WordPress/Jetpack/Resources/release_notes.txt]: total nitpick / personal take: to emphasize that I've checked the WP vs JP release note manually and that I've noted that this is Jetpack-only and removed it from the WordPress notes, as well as not make Grace wonder what this is about (i.e. wonder if this is about Jetpack-the-plugin and not Jetpack-the-app like it actually is), I usually remove the prefix from the Jetpack-specific release notes file at that point 🙃 

```suggestion
* [*] Block Editor: Makes some small changes to the editor's accent colours for consistency. [#19113]
```

OTOH I can see value in keeping this here especially to make it visible in the PR diff, so that reviewers like me think about checking that you indeed removed that entry from the WP release notes file (which you did) 🤔 🤷

**Review by AliSoftware — APPROVED** [review]
- [ ] Pods updated to stable version
   - ⚠️ I'm wondering if you forgot to update `WordPressAuthenticator` to `2.1.0`? (see inline comment)
- [x] New header in `RELEASE-NOTES.txt`
- [x] JP and WP `release_notes.txt` updated with draft notes
- [x] JP release notes audited for differences between WP and JP
- [x] `WordPress/Resources/en.lproj/Localizable.strings` updated with latest strings
- [x] Version bump in `.xcconfig` files

Approving to unblock regardless of this the WordPressAuthenticator version was expected or not, because the beta 1 was already built anyway (so if you need to fix that WPA version you'll have to do a new beta and a new PR anyway 😛 )

File: Podfile.lock
Comment (mokagio) [reply Podfile.lock:606]: Good question. I did create that authenticator new version and noticed it was prompted by changes used in WooCommerce.

I run `bundle exec pod install`, which, correctly, didn't update the version.

I run into a similar case in a past sprint, where I released 2.0.1 but didn't update here. My rationale, in both instances, is that the changes looked WooCommerce-specific.

Now that you call this out, I wonder whether I've been to cautious. The whole point of using SemVer is to leverage it to get upstream compatible updates. There might be new features in a release, but if the coding has been done properly, they should not affect the working of the app.

I'll open a PR against the release branch to update the library, and go from there.

Comment (mokagio) [reply Podfile.lock:606]: > I'll open a PR against the release branch to update the library, and go from there.

See https://github.com/wordpress-mobile/WordPress-iOS/pull/19163. I haven't made it ready for review yet because I wanted to check the auth screen with both 2.0.1 and 2.1.1.

Comment (AliSoftware) [reply Podfile.lock:606]: > I run into a similar case in a past sprint, where I released 2.0.1 but didn't update here. My rationale, in both instances, is that the changes looked WooCommerce-specific.
>
> Now that you call this out, I wonder whether I've been to cautious. The whole point of using SemVer is to leverage it to get upstream compatible updates. There might be new features in a release, but if the coding has been done properly, they should not affect the working of the app.

We have a similar problem with the FluxC library used in both WPAndroid and WCAndroid. Ideally if we were sure that the new versions of the library did follow SemVer, that wouldn't really be an issue. But in practice we don't maintain a CHANGELOG for those libs, nor do we have a tool to compare API changes like [`swift-package-diagnose-api-breaking-changes`](https://forums.swift.org/t/swift-package-diagnose-api-breaking-changes/54308). Which means that when we do a new version, it's hard for the release manager to know if the changes on the lib since last version were breaking changes or minor or patch bumps. Which means we're not guaranteed to follow SemVer in practice, and that a new version `2.1.0`, despite being a minor bump, could totally have broken something in WPiOS by accident compared to its `2.0.0`.

We should probably at some point maintain a `CHANGELOG.md` similar to [the one we use in the release-toolkit](https://github.com/wordpress-mobile/release-toolkit/blob/trunk/CHANGELOG.md), with explicit sections for what changes in the lib are breaking vs backwards-compatible additions vs bug fixes, so that we could garantee respecting SemVer when we pick a new version number, and thus avoid having to ask the question of whether it's safe for our client apps to update to a new release of the lib like here.

Comment (mokagio) [reply Podfile.lock:606]: > We should probably at some point maintain a `CHANGELOG.md`... 
💯

**Pull request merged**
